x-pack/plugins/security/server/routes/user_profile/get_current.ts

```diff
@@ -8,48 +8,48 @@
 import { schema } from '@kbn/config-schema';
 
 import type { RouteDefinitionParams } from '..';
-import type { GetUserProfileResponse } from '../../../common';
+import type { GetUserProfileResponse, UserProfileWithSecurity } from '../../../common';
 import { wrapIntoCustomErrorResponse } from '../../errors';
-import { getPrintableSessionId } from '../../session_management';
 import { createLicensedRouteHandler } from '../licensed_route_handler';
 
-export function defineGetUserProfileRoute({
+export function defineGetCurrentUserProfileRoute({
   router,
-  getSession,
   getUserProfileService,
-  logger,
+  getAuthenticationService,
 }: RouteDefinitionParams) {
   router.get(
     {
       path: '/internal/security/user_profile',
       validate: {
-        query: schema.object({ data: schema.maybe(schema.string()) }),
+        query: schema.object({ dataPath: schema.maybe(schema.string()) }),
       },
     },
     createLicensedRouteHandler(async (context, request, response) => {
-      const session = await getSession().get(request);
-      if (!session) {
+      const authenticationService = await getAuthenticationService();
+      const currentUser = authenticationService.getCurrentUser(request);
+      if (!currentUser) {
         return response.notFound();
       }
 
-      if (!session.userProfileId) {
-        logger.warn(
-          `User profile missing from current session. (sid: ${getPrintableSessionId(session.sid)})`
-        );
```

Comment:
__note:__ to reuse `getCurrent` code in multiple places I had to move this logging to the service itself (even though I'd rather see it here for better context).